### PR TITLE
Trigger connector after it is created or edited

### PR DESCRIPTION
**Pull request by AdityaKhatri**

### CODE REVIEW

File: src/hooks/stateManagement.ts [hunk 1]
```diff
@@ -10,42 +10,43 @@ function isCallable<T>(val: T | ((oldVal: T) => T)): val is (oldVal: T) => T {
...
+            const newValues = isDefined(oldValues) ? [...oldValues] : [];
             newValues.splice(index, 1);
             return newValues;
         });
     }, [setValues, keySelector]);
 
     const modifyItem = useCallback((modifiedItemKey: K, modifiedItem: (T | ((oldVal: T) => T))) => {
         setValues((oldValues) => {
-            const index = oldValues.findIndex(item => keySelector(item) === modifiedItemKey);
+            const safeOldValues = oldValues ? [...oldValues] : [];
```
Comment (tnagorra) [inline src/hooks/stateManagement.ts]: suggestion:
```
        setValues((oldValues= []) => {
        -
```

File: src/hooks/stateManagement.ts [hunk 2]
```diff
@@ -10,42 +10,43 @@ function isCallable<T>(val: T | ((oldVal: T) => T)): val is (oldVal: T) => T {
...
+            ...(oldValues || []),
             newItem,
         ]));
     }, [setValues]);
 
     const removeItem = useCallback((key: K) => {
         setValues((oldValues) => {
-            const index = oldValues.findIndex(item => keySelector(item) === key);
-            if (index === -1) {
+            const index = oldValues?.findIndex(item => keySelector(item) === key);
```
Comment (tnagorra) [inline src/hooks/stateManagement.ts]: Suggestion: give oldValues a default value

File: src/hooks/stateManagement.ts [hunk 3]
```diff
@@ -10,42 +10,43 @@ function isCallable<T>(val: T | ((oldVal: T) => T)): val is (oldVal: T) => T {
...
 ): [
     (item: T) => void,
     (key: K) => void,
     (key: K, item: T) => void,
 ] {
     const addItem = useCallback((newItem: T) => {
-        setValues((oldValues: T[]) => ([
-            ...oldValues,
+        setValues(oldValues => ([
+            ...(oldValues || []),
```
Comment (tnagorra) [inline src/hooks/stateManagement.ts]: set defaultvalue for oldValues

**Review by tnagorra — COMMENTED** [review]
Did we fix loading animation problem with polling request?

**Comment by AdityaKhatri** [comment]
> Did we fix loading animation problem with polling request?

@tnagorra yes

**Review by frozenhelium — APPROVED** [review]

**Pull request merged**